ecs-cli/modules/cli/compose/entity/service/service_helper.go

```diff
@@ -41,16 +95,26 @@ func waitForServiceTasks(service *Service, ecsServiceName string) error {
 			"desiredCount": desiredCount,
 			"runningCount": runningCount,
 		}
+
+		// Log information only if things have changed
+		// running count has changed
+		if runningCount != lastRunningCount {
+			lastRunningCount = runningCount
+			lastRunningCountChangedAt = time.Now()
+			log.WithFields(logFields).Info("Describe ECS Service status")
+		}
+
+		// log new service events
+		if len(ecsService.Events) > 0 {
+			logNewServiceEvents(eventsLogged, ecsService.Events, cmdTimestamp)
+		}
+
+		// The deployment was successful
 		if len(ecsService.Deployments) == 1 && desiredCount == runningCount {
 			log.WithFields(logFields).Info("ECS Service has reached a stable state")
-			return true, nil
+			return nil
 		}
+	}
 
-		if retryCount%2 == 0 {
-			log.WithFields(logFields).Info("Describe ECS Service status")
-		} else {
-			log.WithFields(logFields).Debug("Describe ECS Service status")
-		}
-		return false, nil
-	}, service, timeoutMessage, true)
+	return fmt.Errorf("Deployment has not completed: Running count has not changed for %.2f minutes", timeOut)
```

Comment:
Is this going to be an **error**? If it is, can we return nil and log as **warning** message instead?